theano/configdefaults.py

```diff
@@ -1238,6 +1238,29 @@ def is_valid_check_preallocated_output_param(param):
              in_c_key=True)
 
 
+def check_mkl_openmp():
+    if not theano.config.blas.check_openmp:
+        return
+    import os
+    if ('MKL_THREADING_LAYER' in os.environ and
+            os.environ['MKL_THREADING_LAYER'] == 'GNU'):
+        return
+    try:
+        import mkl
+        if '2018' in mkl.get_version_string():
+            raise RuntimeError('To use MKL 2018 with Theano you MUST set "MKL_THREADING_LAYER=GNU" in your environement.')
+    except ImportError:
+        raise RuntimeError("""
```

Comment:
I just want to clarity, with this PR, now the mkl package is mandatory with conda. If the user don't use conda, and the string "mkl" isn't in the flag, then he don't need to change his environment. Can you confirm that?

Reply:
If the string mkl isn't in the flag, then nothing changes.

However if you use mkl and are not in conda then you either need to obtain the mkl-service package somehow, set MKL_THREADING_LAYER=GNU or disable the check with blas.check_openmp=False